polling_stations/apps/data_importers/management/commands/import_eoni_from_s3.py

```diff
@@ -0,0 +1,415 @@
+import csv
+import re
+from pathlib import Path
+
+import chardet
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from pyproj import Transformer
+
+from data_importers.s3wrapper import parse_s3_uri, S3Wrapper
+from data_importers.management.commands.import_eoni import Command as EONI_Importer
+
+from polling_stations.settings.constants.importers import EONIImportScheme
+
+
+def attempt_decode(body: bytes):
+    sample = get_body_sample(body)
+    detection = chardet.detect(sample)
+    encoding = detection["encoding"]
+    if encoding == "utf-16le":
+        return body.decode(encoding), encoding
+    encodings = ["utf-8", "windows-1252", "latin-1"]
+    for encoding in encodings:
+        try:
+            return body.decode(encoding), encoding
+        except UnicodeDecodeError as e:
+            last_exception = e
+            continue
+    raise last_exception
+
+
+def get_body_sample(body: bytes):
+    if b"\r\n" in body:
+        line_sep = b"\r\n"
+    elif b"\n" in body:
+        line_sep = b"\n"
+    else:
+        return body
+    lines = body.split(line_sep)
+    if len(lines) > 20:
+        return line_sep.join(lines[:20])
+    return line_sep.join(lines)
+
+
+def is_eoni_export_key(key: str) -> bool:
+    """
+    Should return true for things like:
+    '2023-04-12-EONIextract.txt',
+    '2024-07-03-EONIextract-sample.txt',
+    '2024-11-21-EONIextract_NEW.txt'
+
+    """
+    eoni_export_key_pattern = re.compile(
+        r"""
+        ^           # Start of string
+        20[2-9]     # First 3 digits of year: 202-209
+        [0-9]       # Last digit of year: 0-9
+        -           # Separator
+        [0-9]{2}    # Month as 2 digits
+        -           # Separator
+        [0-9]{2}    # Day as 2 digits
+        -.+         # Hyphen followed by any text
+        \.txt       # Literal .txt extension
+        """,
+        re.VERBOSE,
+    )
+    if eoni_export_key_pattern.match(key):
+        return True
+    return False
```

Comment:
At the moment the buckets we've used have only been used for EONI exports, which have followed this naming convention. This is as much to document what we expect to find as anything else.